src/community/wfs-templating/src/main/java/org/geoserver/wfstemplating/expressions/XpathFunction.java

```diff
@@ -0,0 +1,67 @@
+package org.geoserver.wfstemplating.expressions;
+
+import static org.geotools.filter.capability.FunctionNameImpl.parameter;
+
+import org.geotools.filter.AttributeExpressionImpl;
+import org.geotools.filter.FunctionExpressionImpl;
+import org.geotools.filter.capability.FunctionNameImpl;
+import org.opengis.filter.capability.FunctionName;
+import org.opengis.filter.expression.ExpressionVisitor;
+import org.opengis.filter.expression.PropertyName;
+import org.xml.sax.helpers.NamespaceSupport;
+
+public class XpathFunction extends FunctionExpressionImpl implements PropertyName {
```

Comment:
A class that's both a Function and a PropertyName is, I believe, something we haven't seen before. Seems pretty similar to "FilterFunction_property" down in Geotools, which also allows to dynamically reference a property, and which has received special treatment to work with the various visitors correctly.
Now, this is just an output format, so there is no data access and the many DuplicatingFilterVisitor subclasses being involved into it. If it actually adds something (namespaces for example?) then I would clearly document it as something specific to this module, that should not be used anywhere else, and point people to "FilterFunction_property" for general usage. As another option, the existing "FilterFunction_property" function could be expanded to support what's missing here.

Reply:
@aaime thanks for your point I've updated the documentation pointing the FilterFunction_porperty